src/gui/userstatus.cpp

```diff
@@ -28,43 +29,50 @@ namespace OCC {
 
 Q_LOGGING_CATEGORY(lcUserStatus, "nextcloud.gui.userstatus", QtInfoMsg)
 
-UserStatus::UserStatus(QObject *parent)
-    : QObject(parent)
-{
-}
-
-UserStatus::Status UserStatus::stringToEnum(const QString &status) const 
-{
-    // it needs to match the Status enum
-    const QHash<QString, Status> preDefinedStatus{{"online", Status::Online},
-                                               {"dnd", Status::DoNotDisturb}, //DoNotDisturb
-                                               {"away", Status::Away},
-                                               {"offline", Status::Offline},
-                                               {"invisible", Status::Invisible}};
+namespace {
+    UserStatus::Status stringToEnum(const QString &status) 
+    {
+        // it needs to match the Status enum
+        const QHash<QString, UserStatus::Status> preDefinedStatus{
+            {"online", UserStatus::Status::Online},
+            {"dnd", UserStatus::Status::DoNotDisturb},
+            {"away", UserStatus::Status::Away},
+            {"offline", UserStatus::Status::Offline},
+            {"invisible", UserStatus::Status::Invisible}
+        };
+        
+        // api should return invisible, dnd,... toLower() it is to make sure 
+        // it matches _preDefinedStatus, otherwise the default is online (0)
+        return preDefinedStatus.value(status.toLower(), UserStatus::Status::Online);
+    }
     
-    // api should return invisible, dnd,... toLower() it is to make sure 
-    // it matches _preDefinedStatus, otherwise the default is online (0)
-    const auto statusKey = status.isEmpty() ? QStringLiteral("online") : status.toLower();
-    return preDefinedStatus.value(statusKey, Status::Online);
+    QString enumToString(UserStatus::Status status) 
+    {
+        switch (status) {
+        case UserStatus::Status::Away:
+            return QObject::tr("Away");
+        case UserStatus::Status::DoNotDisturb:
+            return QObject::tr("Do not disturb");
+        case UserStatus::Status::Invisible:
+        case UserStatus::Status::Offline:
+            return QObject::tr("Offline");
+        default:
```

Comment:
maybe it is me but I tend to not use `default` because it will silently accept any extra value added in the future without a clue at compilation
why do you keep it instead of just using a `case UserStatus::Status::Online:` ?

Reply:
@mgallien I guess in that case we'd still need `default:` to return something. Otherwise, this may not compile due to not every code path having a return value.  Adding a return after the `switch` body may work, but, there'd still be a warning during the compilation.

Reply:
@allexzander I agree with @mgallien, and I think you can use `Q_UNREACHABLE` for example for these cases. If you don't want to have something hard, return something like `"Unsupported status"` by default.

Reply:
So we do not use `default` especially in those kind of cases and `Q_UNREACHABLE();` is needed after that to prevent problems and prevent warnings.